website/static/js/share/index.js

```diff
@@ -102,15 +116,39 @@ ShareApp.controller = function() {
     });
 };
 
-var renderSort = function(ctrl){
+var SortBox = {
+    view: function(ctrl, params) {
+        var vm = params.vm;
         return m('.btn-group.pull-right', [
             m('button.btn.btn-default.dropdown-toggle', {
                     'data-toggle': 'dropdown',
                     'aria-expanded': 'false'
-                }, ['Sort by: ' + ctrl.vm.sort() + ' ', m('span.caret')]
+                }, ['Sort by: ' + vm.sort() + ' ', m('span.caret')]
             ),
-                m('ul.dropdown-menu', {'role': 'menu'}, ctrl.sideBarController.renderSort())]);
-    };
+                m('ul.dropdown-menu', {'role': 'menu'},
+                    $.map(Object.keys(vm.sortMap), function(a) {
+                        return m.component(SortItem, {vm: vm, item: a});
+                    })
+                )
+        ]);
+
+    }
+};
 
+var SortItem = {
+    view: function(ctrl, params) {
+        var vm = params.vm;
+        var item = params.item;
+        return m('li',
+            m('a', {
```

Comment:
Using anchor tags that point to `#` makes the page jump to the top, which hurts usability.